docs/howtos/lr_schedule.rst

```diff
@@ -0,0 +1,74 @@
+Managing Parameters and State
+=============================
+
+We will show you how to...
+
+* define a learning rate schedule
+* train a simple model using that schedule
+
+.. code-block:: python
+  
+  def create_triangular_schedule(lr_min, lr_max, steps_per_cycle):
+  """Return fn from epoch to LR, linearly interpolating between `lr_min`->`lr_max`->`lr_min`."""
+    top = (steps_per_cycle + 1) // 2
+    def learning_rate_fn(step):
+      cycle_step = step % steps_per_cycle
+      if cycle_step < top:
+        lr = lr_min + cycle_step/top * (lr_max - lr_min)
+      else:
+        lr = lr_max - ((cycle_step - top)/top) * (lr_max - lr_min)
+      return lr
+    return learning_rate_fn
+
+This example shows how to create a simple cyclical learning rate scheduler, the triangular scheduler.
+
+To use the schedule one must simply create a learning rate function by passing the hyperparameters to the 
+create_triangular_schedule function and then use that function to compute the learning rate for your updates.
+For example using this schedule on MNIST would require changing the train_step function
+
+.. code-block:: python
+  
+  def train_step(optimizer, batch, learning_rate_fn):  
+    def loss_fn(params):
+      logits = CNN().apply({'params': params}, batch['image'])
+      loss = cross_entropy_loss(logits, batch['label'])
+      return loss, logits
+    
+    grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+    (_, logits), grad = grad_fn(optimizer.target)
+    step = optimizer.state.step
+    lr = learning_rate_fn(step)
+    optimizer = optimizer.apply_gradient(grad, {"learning_rate":lr})
+    metrics = compute_metrics(logits, batch['label'])
+    return optimizer, metrics
```

Comment:
This is a great use-case for our brand new Sphinx codediff directive! It allows you to display code in side-by-side view. Can you check how this looks if you compile it locally?

```suggestion
.. codediff::
  :title_left: Default learning rate
  :title_right: Triangular learning rate schedule
  
  @jax.jit
  def train_step(optimizer, batch): #!
    def loss_fn(params):
      logits = CNN().apply({'params': params}, batch['image'])
      loss = cross_entropy_loss(logits, batch['label'])
      return loss, logits
    grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
    (_, logits), grad = grad_fn(optimizer.target)


    optimizer = optimizer.apply_gradient(grad) #!
    metrics = compute_metrics(logits, batch['label'])
    return optimizer, metrics
  ---
  @jax.jit
  def train_step(optimizer, batch, learning_rate_fn): #!
    def loss_fn(params):
      logits = CNN().apply({'params': params}, batch['image'])
      loss = cross_entropy_loss(logits, batch['label'])
      return loss, logits
    grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
    (_, logits), grad = grad_fn(optimizer.target)
    step = optimizer.state.step #!
    lr = learning_rate_fn(step) #!
    optimizer = optimizer.apply_gradient(grad, {"learning_rate":lr}) #!
    metrics = compute_metrics(logits, batch['label'])
    return optimizer, metrics
```

Reply:
It looks good, though do you know if there's a way to align the top of the code-boxes?

Reply:
They should be align to the top by default. Did you sync to head? I made some changes to the CSS recently so many they aren't yet part of your change?

Reply:
Ah sorry I see those changes are not merged yet, they are actually part of #985 . I'll split off those CSS changes and get them merged so you can sync to head.

Reply:
I just merged #990, could you sync to head and let me know how it looks now?

Reply:
Looks great now!